internal/db/secrets.go

```diff
@@ -126,6 +126,9 @@ func (d *Database) Get(ctx context.Context, tag names.CloudCredentialTag) (_ map
 	secret := dbmodel.NewSecret(tag.Kind(), tag.String(), nil)
 	err = d.GetSecret(ctx, &secret)
 	if err != nil {
+		if errors.ErrorCode(err) == errors.CodeNotFound {
+			return nil, nil
```

Comment:
why are we returning no error if it is not found?

Reply:
This is the behaviour of our Vault client, that test where I set it to run against Postgres and Vault showed that our Postgres implementation was not consistent with our Vault implementation. So I've changed the Postgres implementation to match Vault.